hazelcast-spring/src/main/resources/hazelcast-spring-4.0.xsd

```diff
@@ -3257,6 +3257,16 @@
                 </xs:documentation>
             </xs:annotation>
         </xs:attribute>
+        <xs:attribute name="epochStart" type="xs:long" default="1514764800000" use="optional">
+            <xs:annotation>
+                <xs:documentation>
+                    Sets the offset of timestamp component. Time unit is milliseconds, default is 1514764800000 
+                    (1.1.2018 0:00 UTC).
+
+                    This setting pertains only to newId() calls made on the member that configured it.
+                </xs:documentation>
+            </xs:annotation>
+        </xs:attribute>
         <xs:attribute name="idOffset" type="xs:long" default="0" use="optional">
```

Comment:
I think we can remove this setting now. It can be used to similar effect as `epochStart` and was designed to ease migration from the old `IdGenerator`. The `IdGenerator` is now removed anyway.

Reply:
There are lots of `idOffset` dependencies from Hazelcast Client Protocol. Does it support field removal?

Reply:
Protocol does not support field removal. But in 4.0, we can change the protocol, since 4.0 is not backward compatible.

Reply:
Ok, covered in https://github.com/hazelcast/hazelcast-client-protocol/pull/281/commits/faf444101f17ee7603326b6615281586e8149b82.

Reply:
There's still a lot of id-offset related code left... I'm going to remove it.

Reply:
Please recheck the most recent commit.